EIPS/eip-223.md

```diff
@@ -0,0 +1,389 @@
+---
+eip: 223
+title: 223 Token with communication model
+description: Token with event handling and communication model
+author: Dexaran (@Dexaran)
+discussions-to: https://ethereum-magicians.org/t/erc-223-token-standard/12894
+status: Review
+type: Standards Track
+category: ERC
+created: 2017-05-03
+---
+
+## Abstract
+
+The following describes standard functions a token contract and contract which is intended to work with specified token can implement. This standard introduces a communication model which allows for the implementation of **event handling** on the receiver's side.
+
+## Motivation
+
+1. This token introduces a communication model for contracts that can be utilized to straighten the behavior of contracts that interact with tokens as opposed to [ERC-20](./eip-20.md) where a token transfer could be ignored by the receiving contract.
+2. This token is more gas-efficient when depositing tokens to contracts.
+3. This token allows for `_data` recording for financial transfers.
+
+
+## Specification
+
+Token
+Contracts that works with tokens
+
+### Methods
+
+NOTE: An important point is that contract developers must implement `tokenReceived` if they want their contracts to work with the specified tokens.
+
+If the receiver does not implement the `tokenReceived` function, consider the contract is not designed to work with tokens, then the transaction must fail and no tokens will be transferred. An analogy with an Ether transaction that is failing when trying to send Ether to a contract that did not implement `receive() payable`.
+
+
+#### totalSupply
+
+```js
+
+function totalSupply() constant returns (uint256 totalSupply)
+
+```
```

Comment:
````suggestion
```solidity
function totalSupply() constant returns (uint256 totalSupply)
```
````

---

On another note, I'm not familiar with the `constant` function modifier. Would that be `pure`/`view` in modern Solidity?

Reply:
Yes, I'll update the code description to match the current version of Solidity

Here is the actual 0.8.0 code https://github.com/Dexaran/ERC223-token-standard/blob/development/token/ERC223/ERC223.sol